netket/graph/symmetry.py

```diff
@@ -80,6 +80,49 @@ def remove_duplicates(self, *, return_inverse=False):
         else:
             return group
 
+    def inverse(self):
+
+        """
+        Computes the inverse permutatations of automorphisms s.t. for each element,
+        applying inverse to automorphisms produces the identity
+
+        """
+
+        automorphisms = self.to_array()
+        n_symm = len(automorphisms)
+        inverse = np.zeros([n_symm], dtype=int)
+        automorphisms = np.array(automorphisms)
+        for i, perm1 in enumerate(automorphisms):
+            for j, perm2 in enumerate(automorphisms):
+                perm_sq = perm1[perm2]
+                if np.all(perm_sq == np.arange(len(perm_sq))):
+                    inverse[i] = j
+
+        return automorphisms[inverse]
+
+    def group_algebra(self):
+
+        """
+        Computes an array relative displacements between poses in automorphisms, s.t.
+        group_algebra[i,j] = inverse[i]*automorphisms[j]
+
+        This array is flattened and converted to a tuple before returning
+
+        """
+
+        automorphisms = self.to_array()
+        n_symm = len(automorphisms)
+        inverse = self.inverse()
+        group_algebra = np.zeros([n_symm, n_symm], dtype=int)
+
+        for i, inv in enumerate(inverse):
+            for j, perm in enumerate(automorphisms):
+                for k, filter in enumerate(automorphisms):
+                    if np.all(perm[inv] == filter):
+                        group_algebra[i, j] = k
+
+        return tuple(group_algebra.ravel())
```

Comment:
Why is this converted to a tuple? Is this in order to work with Flax?

Reply:
Yeah, I've fixed it and made it a HashableArray

Reply:
Great. I just would like @femtobit opinion here about whoever it makes sense to return an hashable array here or if we should keep the build_GCN method and return a normal array here...

I think that if we make HashableArray behave satisfy the numpy API we might get away with just returning a HashableArray. 

This means adding dtype,shape,__len__,ndim,getitem I think...

Then you could use it as a numpy array and all is good as long as you don't modify it.

Reply:
> I think that if we make HashableArray behave satisfy the numpy API we might get away with just returning a HashableArray.

I wanted to avoid that. Making it behave like a numpy array _most of the time_ is doable, but there might be additional bugs or at least inconsistencies when it doesn't. (Just as `DeviceArray` and `ndarray` mostly behave the same but have subtle differences.)

The only time you need a `HashableArray` is when you use it as a model attribute. I feel we should just wrap it there and use normal objects otherwise.

Reply:
Maybe it also makes sense for group algebra to be its own class? I haven't had time to look at this PR in more detail so I don't yet know if that makes any sense though. I'll take a closer look in the next days.

Reply:
I moved group_algebra to be a function in SymmGroup. I don't see why you would need to call it on something that isn't a SymmGroup object